beacon-chain/p2p/discovery.go

```diff
@@ -122,29 +125,47 @@ func (s *Service) listenForNewNodes() {
 			time.Sleep(pollingPeriod)
 			continue
 		}
-
-		if exists := iterator.Next(); !exists {
-			break
-		}
-
-		node := iterator.Node()
-		peerInfo, _, err := convertToAddrInfo(node)
-		if err != nil {
-			log.WithError(err).Error("Could not convert to peer info")
+		wantedCount := s.wantedPeerDials()
+		if wantedCount == 0 {
+			log.Trace("Not looking for peers, at peer limit")
+			time.Sleep(pollingPeriod)
 			continue
 		}
-
-		if peerInfo == nil {
-			continue
+		// Restrict dials if limit is applied.
+		if flags.MaxDialIsActive() {
+			var err error
+			wantedICount := math.Min(uint64(wantedCount), uint64(flags.Get().MaxConcurrentDials))
+			wantedCount, err = math.Int(wantedICount)
```

Comment:
Here you convert both `wantedCount` and `MaxConcurrentDials` which are `int` to `uint64`, then get the `min`, then convert them back to `int`.

Could be done in one step with the builtin `min` function.